internal/rest/rest.go

```diff
@@ -52,9 +57,76 @@ func handleAPIRequest(action rest.EndpointAction, state *internalState.State, w
 		}
 	}
 
+	if action.ProxyTarget {
+		return proxyTarget(action, state, r)
+	}
+
 	return action.Handler(state, r)
 }
 
+func proxyTarget(action rest.EndpointAction, s *internalState.State, r *http.Request) response.Response {
+	if !s.Database.IsOpen() {
+		return response.Unavailable(fmt.Errorf("Cannot send request to target. Daemon not yet initialized"))
+	}
+
+	if r.URL == nil {
+		return action.Handler(s, r)
+	}
+
+	values, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		logger.Warnf("Failed to parse query string %q: %v", r.URL.RawQuery, err)
```

Comment:
I noticed in LXD, we don't error out if we fail to parse a query, instead just returning a `Warn` log and ignoring the query. I kept the same behaviour, but I'm not sure why this is the case. Thoughts @stgraber?

Reply:
Not exactly sure why we have that in LXD.

Reply:
In LXD that logic is inside the `queryParam(request *http.Request, key string) string` helper which doesn't return an error for convenience (of not having to check it). The query param is either there or not, and if the query string is invalid, its not there effectively.

In this case I think its probably more correct to return an error if an invalid query string is passed.